primitiv/parameter.h

```diff
@@ -65,27 +44,57 @@ class Parameter : mixins::Noncopyable<Parameter> {
    */
   Parameter(
       const Shape &shape,
-      const Initializer &init,
+      const Initializer &initializer,
       Device &device = Device::get_default());
 
   /**
-   * Returns whether the parameter is valid or not.
-   * @return true or false w.r.t. the parameter is valid or not.
+   * Initializes the Parameter object.
+   * @param shape The shape of the parameter. The batch size should be 1.
+   * @param value List of initial values. Order of elements should be the
+   *              column-major (Fortran) order.
+   * @param device The device object to manage internal memory.
    */
-  bool valid() const { return !!device_; }
+  void init(
+      const Shape &shape,
+      const std::vector<float> &value,
+      Device &device = Device::get_default());
+
+  /**
+   * Initializes the Parameter object.
+   * @param shape The shape of the parameter. The batch size should be 1.
+   * @param init An Initializer object.
+   * @param device The device object to manage internal memory.
+   */
+  void init(
+      const Shape &shape,
+      const Initializer &initializer,
+      Device &device = Device::get_default());
 
```

Comment:
Some functions take `shape` and `initializer` in this order in C++, whereas python-primitiv can also take an `np.ndarray` argument without a `shape` argument.
Therefore, it is very useful if the order is `initializer` -> `shape`.

Do you have a plan to change this order?

Reply:
It is better to move this topic to other issue.

Reply:
See #57.

Reply:
Okay. For now, I use keyword arguments with the same order.